src/main/java/com/passbook/sparkeighteen/service/TransactionService.java

```diff
@@ -53,4 +63,59 @@ private Float getUpdatedBalance(UserEntity user) {
         // TODO: Add getting the latest closing balance of the user
         return balance;
     }
+
+    /**
+     * User gets all transaction list in paginated format.
+     * @param userID is used to show all user performing transaction of that particular user.
+     * @param pageSize is basically used to give the size of the page. user can set a page size to show how many transaction want to show you on a page.
+     * @return the user transaction list by Paginated.
+     */
+    public PaginatedResponse getTransaction(String pageNo, String pageSize, @NotNull Integer userID) {
+
+        final Optional<UserEntity> optionalUser = userRepository.findById(userID);
+        if (optionalUser.isEmpty())
+            return PaginatedResponse.builder()
+                    .message("user not found")
+                    .build();
+
+        Pageable paging =
+                PageRequest.of(
+                        Integer.parseInt(pageNo),
+                        Integer.parseInt(pageSize),
+                        Sort.by("time").ascending());
+        Page<TransactionEntity> transactions =
+                transactionRepository.findByUser(paging, optionalUser.get());
+
+        if (transactions.isEmpty())
+            return PaginatedResponse.builder()
+                    .message("No transactions for this user.")
+                    .build();
+
+        List<TransactionResponse> response =
+                transactions.getContent().stream()
+                        .map(this::toTransactionResponse)
+                        .collect(Collectors.toList());
+        return PaginatedResponse.builder()
+                .content(Collections.singletonList(response))
```

Comment:
this wrapper might not be needed, if generics are used properly in PaginatedResponse class content field.